modules/composer-ffmpeg/src/main/java/org/opencastproject/composer/impl/ComposerServiceImpl.java

```diff
@@ -2185,6 +2194,85 @@ private SmilMediaParamGroup getSmilMediaParamGroup(Smil smil, String trackParamG
     }
   }
 
+  @Override
+  public Job multiEncode(Track sourceTrack, List<String> profileIds) throws EncoderException, MediaPackageException {
+    try {
+      // Job Load is based on number of encoding profiles
+      float load = calculateJobLoadForMultipleProfiles(profileIds, multiEncodeJobLoadFactor);
+      ArrayList<String> args = new ArrayList<String>();
+      args.addAll(Arrays.asList(MediaPackageElementParser.getAsXml(sourceTrack)));
+      args.addAll(profileIds);
+      return serviceRegistry.createJob(JOB_TYPE, Operation.MultiEncode.toString(), args, load);
+    } catch (ServiceRegistryException e) {
+      throw new EncoderException("Unable to create a job", e);
+    }
+  }
+
+  /**
+   * A single encoding process that produces multiple outputs from a single track(s) using a list of encoding profiles.
+   * Each output can be tagged by the profile name.
+   *
+   * @param job
+   *          - encoding job
+   * @param track
+   *          - source track
+   * @param profileIds
+   *          - list of encoding profile Ids
+   * @param properties
+   *          - used in the command templates from profiles
+   * @return encoded files
+   * @throws EncoderException
+   *           - if can't encode
+   * @throws IllegalArgumentException
+   *           - if missing arguments
+   *
+   */
+  protected List<Track> multiEncode(final Job job, Track track, List<String> profileIds,
+          Map<String, String> properties) throws EncoderException, IllegalArgumentException {
+    if (job == null)
+      throw new IllegalArgumentException("The Job parameter must not be null");
+    if (track == null)
+      throw new IllegalArgumentException("Source track cannot be null");
+    if (profileIds == null || profileIds.isEmpty())
+      throw new IllegalArgumentException("Cannot encode without encodeing profiles");
+    try {
+        final File videoFile = loadTrackIntoWorkspace(job, "source", track);
+        // Get the encoding profiles
+        List<EncodingProfile> profiles = new ArrayList<EncodingProfile>();
+        for (String profileId : profileIds) {
+        EncodingProfile profile = getProfile(job, profileId);
+        profiles.add(profile);
+      }
+      logger.info("Encoding source track {} using profiles '{}'", track.getIdentifier(), profileIds);
+      // Do the work
+      EncoderEngine encoderEngine = getEncoderEngine();
+      List<File> outputs;
+      try {
+        outputs = encoderEngine.multiTrimConcat(Arrays.asList(videoFile), null, profiles, 0, track.hasVideo(),
+                track.hasAudio());
+      } catch (EncoderException e) {
+        Map<String, String> params = new HashMap<>();
+        List<String> profileList = new ArrayList<>();
```

Comment:
This is updated two lines below but never used anywhere, is it?

Reply:
removed